src/equation_systems/equation_system.C

```diff
@@ -106,12 +106,53 @@ EquationSystem::ApplyBoundaryConditions(platypus::BCMap & bc_map)
         test_var_name, _lfs.GetRef(test_var_name), _test_pfespaces.at(i)->GetParMesh());
   }
 }
+
 void
 EquationSystem::FormLinearSystem(mfem::OperatorHandle & op,
                                  mfem::BlockVector & trueX,
                                  mfem::BlockVector & trueRHS)
 {
 
+  if (_assembly_level == mfem::AssemblyLevel::LEGACY)
+  {
```

Comment:
Switch case here will be easier to extend as more supported assembly levels are added

Reply:
Yeah, that's a good point. Fixed!